### PR TITLE
Success/error status issue

### LINKED ISSUE #73: Indicate to users that their submissions have been sent
Right now the dialog just closes when a user adds a clinic or reports one. There's nothing that indicates that their request went through. We should add a small notification somewhere on the screen.

### PR DESCRIPTION
**Pull request by amar-prabhu-29**
## Resolves  #73
Resolved Issue #73 by added a new dialogue which is triggered when  a user completes an action (Form Submission)
The dialogue is reusable and the same can be used for any future dialogue requirements in case of success or error. 

### How to test
Tests
1) Submit the add clinic form
2) Report a clinic

### Screenshots
Capture

# Documentation
Documentation for file : components/StatusModal.vue
This is a dialogue or modal component to show status Success or error preferably when 
a user does a action such as submitting a form 

## Props
title : The Title of the dialogue (Modal) preferably "Success" or "Error"
text: The paragraph which gives more information about the status.

## Events
Closed: This event is emitted from the component when the user clicks Action Button Ok.

## Example
the following snippet can be used in template to initialize the model with appropriate data.
```
<StatusModal ref="successDialog" title="Success" text="New clinic data successfully inserted to the database." @closed="$router.push('/')"/>
```
The following snippet can be added in any method to trigger the dialog to open
```
this.$refs.successDialog.open();
```

### CODE REVIEW

File: components/CancellationList.vue
Comment (Ermish) [inline components/CancellationList.vue:79]: For Vue, it's typically a better practice to use `data` or `props` for reactive properties instead of `$ref`s. 

For this specific case, I would recommend passing in a `prop` from the parent. 

See[ this example](https://github.com/ourjapanlife/findadoc-frontend/blob/fad247b07039a55bf7cfd6da54b248c9f624b995/components/PendingSubmissions.vue#L5) of `deleteDialog` being used in `PendingSubmissions` parent component.

File: components/StatusModal.vue
Comment (Ermish) [inline components/StatusModal.vue:3]: For `Boolean` properties, it's a good practice to prefix with a verb. Like `showStatusDialog` or `isStatusDialogOpen`

Comment (amar-prabhu-29) [reply components/StatusModal.vue:3]: Yes understood, ill do the required changes and commit.

File: components/StatusModal.vue
Comment (Ermish) [inline components/StatusModal.vue:15]: There's actually some unofficial naming conventions for event handling!

### For events
the `action` itself is fine. `click`, `tapped`, `tap`, `scrollup`, etc.  (Personally I prefer past tense....`clicked` vs. `click`.

### For event handlers
You prefix the event with `handle` like `handleClicked` or `handleClosed`.

### For event hooks/functions props called when an event happens
You prefix the event with `on` like `onClicked` or `onClosed`.

#### How are event hooks/ function props different than a handler? 
Honestly, these are often the same thing, but sometimes they are distinct. 
They are typically delegate methods that might be run when the event happens but don't guarantee the event has been "handled".  Attaching logging is a good example

cc @theyokohamalife

File: pages/add-clinic.vue
Comment (Ermish) [inline pages/add-clinic.vue:138]: instead of `then()` let's use the `async/await` pattern. 

slack me if you need help and I can pair with you :)

**Review by Ermish — CHANGES_REQUESTED** [review]
Requested a few changes, but overall looks good!

**Comment by PsychicNoodles** [comment]
As an FYI, there are existing Vue/Nuxt libraries for popups like this, such as [@nuxt/toast](https://github.com/nuxt-community/community-modules/tree/master/packages/toast), and using an existing library may be easier to maintain and probably offers more options. That said, I don't think there's any particular problem using a custom solution if everyone's okay with it!

**Comment by amar-prabhu-29** [comment]
I was trying to create a reusable dialog box to be used across all form submissions or any other case where a dialog maybe necessary so we are consistent with the design/look and feel of the dialog. 

 Is it better to have a existing library for the same?
 
 What do y'all think?

**Comment by PsychicNoodles** [comment]
We definitely would want a consistent UI for displaying status messages. I'm not a design expert by any means so unless there are any around it will probably just come down to personal preferences

I think for simple messages like "submitted successfully" or "login failed" it might be easier to choose an existing library to reduce the chances of making mistakes (speaking from experience of trying to replicate seemingly simple stuff and ending up spending hours on it), but I'll defer to the full on maintainers here.

**Comment by Ermish** [comment]
@PsychicNoodles is right! We need a consistent experience and toast notifications would be great to have. I’ve created #103 to set them up.
 @theyokohamalife let’s talk about our library options!

Cc @amar-prabhu-29

**Pull request closed without merging**
